service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java

```diff
@@ -25,42 +25,49 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(PayonePostServiceImpl.class);
     private static final String ENCODING_UTF8 = "UTF-8";
+    private final List<String> personalDataToRemove;
 
     private String serverAPIURL;
 
     private int connectionTimeout = 3000;
 
 
-    private PayonePostServiceImpl(final String serverAPIURL) {
+    private PayonePostServiceImpl(final String serverAPIURL,
+                                  final List<String> personalDataToRemove) {
         Preconditions.checkArgument(
                 serverAPIURL != null && !serverAPIURL.isEmpty(),
                 "The server api url must not be null or empty.");
+        Preconditions.checkArgument(personalDataToRemove != null,
+                "personalDataToRemove cannot be null");
```

Comment:
Why not use good default (set to true) instead of making it mandatory and throwing an error?

Reply:
It is a string

Reply:
I think it is not really user friendly to throw an error and let the user figure out on version update which fields are logged and qualify as privat or not as otherwise an NPE will be thrown. We need to define default fields here as constant if `personalDataToRemove` is not defined and check if making them configurable does bring ans value or just complexity. Just for debugging issues (if necessary at all as un-anonymised data is still available on both customer and cart objects) one could make `personalDataToRemove` as boolean which will just activate or deactivate anonymisation.